### ISSUE TITLE
javalite.zip 

### ISSUE THREAD

**Issue by plaird** [protocolbuffers/protobuf#6561]
Our product build (implemented in Bazel) was relying on a javalite.zip with a particular SHA. Sadly, we don't have a local copy that we can find (we thought it would be in our local repository cache, but no....). It seems that the file was just replaced with a newer one, but the newer one breaks our build.

Is there anyway to retrieve the older copy? This is our WORKSPACE snippet:

http_archive(
    name = "com_google_protobuf_javalite",
    strip_prefix = "protobuf-javalite",
    urls = ["https://github.com/google/protobuf/archive/javalite.zip"],
    sha256 = "e1c862a1a32e19e39ef15c7c9fe1aaedd3fee5a76967370a028b786f2a186945",
)

**Comment by acozzette** [comment]
It shouldn't be too hard to figure out which commit it was from. If you look in your repo history to see when that sha256 was last updated, you can then look in the protobuf repo history to see what commit was at the head of the `javalite` branch at that same time. Then update the URL to `https://github.com/google/protobuf/archive/<the Git commit>.zip`.

**Comment by btbuxton** [comment]
The URL gets the protobuf version, how can we get the javalite version that goes it? For example, with version 3.6.1.3? Are javalite versions kept as well? @acozzette

**Comment by acozzette** [comment]
Java lite didn't get any releases for a while, because we had to make some ABI-incompatible changes to it. We started publishing lite releases again with 3.8.0, though.

**Comment by btbuxton** [comment]
how would I get the 3.8.0 version from the archive? I tried javalite-v3.8.0.zip and javalite-09745575a923640154bcf307fba8aedff47f240a.zip. Is the only way to get javalite.zip is to get the latest?

**Comment by acozzette** [comment]
Yes, sorry for the trouble but I think at the moment the only way to refer to Java lite with Bazel is to specify the Git commit you want, or point to the branch so that you always get the latest. CC @rafi-kamal who might know if there is a better way.

**Comment by plaird** [comment]
Thanks Adam! With your hints we got a solution. There was this commit two days ago on the javalite branch:

"Prepare for Bazel-incompatible changes"
https://github.com/protocolbuffers/protobuf/commit/55735abe1ab010f9943c750b507c3ae6b3740437

The commit prior is 0ebb905ff877119d2f0683f968f0a104a54fd0d1. Given that, we converted our Bazel WORKSPACE to use the git_repository rule at that commit instead of http_archive (using latest):

git_repository(
    name = "com_google_protobuf_javalite",
    remote = "https://github.com/google/protobuf",
    commit = "0ebb905ff877119d2f0683f968f0a104a54fd0d1",
)

Boom. Problem solved. Thanks again for the FAST responses.
